sysdeps/linux/generic/sysdeps.cpp

```diff
@@ -271,21 +271,21 @@ int sys_sleep(time_t *secs, long *nanos) {
 	return 0;
 }
 
+int sys_isatty(int fd) {
+	struct winsize ws;
+	auto ret = do_syscall(NR_ioctl, fd, TIOCGWINSZ, &ws);
```

Comment:
This doesn't work. It relies on ``ioctl()`` numbers which are not available in ansi-only. But doing this isn't necessary, your test shouldn't be failing because of ``sys_isatty()`` missing, and if it is then that is a separate issue that needs to be addressed properly.